### ISSUE TITLE
rabbitmq  container run error in eShopLite sample

### ISSUE THREAD

**Issue by salarvand** [dotnet/aspire#995]
throw exception when run `var messaging = builder.AddRabbitMQContainer("messaging");` in eShopLite sample and do not create container.

**Comment by davidfowl** [comment]
Can you show the exception? Or what errors you are running into.

**Comment by salarvand** [comment]
Exactlly in `claass AspireRabbitMQExtensions`  and method :` private static IConnection CreateConnection(IConnectionFactory factory, int retryCount) `
when wanted to CreateConnection : 
```
try
{
    return factory.CreateConnection();
}
catch (Exception ex)
```

But when you create rabbitmq container manually in docker desktop with name of "messaging" every thing is ok ! the main problem is Aspire.Hostin can not create rabbitmq container automatically but for redis and postgreSQL it is every thing ok.

Can you address me in source code where Aspire create containers for docker desktop from image names ?

**Comment by davidfowl** [comment]
Can you show me the entire output? I’m still not sure what’s happening. What does the dashboard say about the rabbitmq container? What about logs from the dashboard?

**Comment by salarvand** [comment]
I can't run project to show you dashboard's logs ! but I send you an image with exception details. I would be grateful if you could tell me how do Aspire create containers for docker desktop from image names (in source code) ?

Exception Detail :  RabbitMQ.Client.Exceptions.BrokerUnreachableException: 'None of the specified endpoints were reachable'
Inner Exception : IOException: connection.start was never received, likely due to a network timeout


![Error](https://github.com/dotnet/aspire/assets/8214930/3b924dff-05a1-4c9c-8829-16b650988c03)

**Comment by davidfowl** [comment]
This is a normal exception, that should get caught and retired. If you continue does to eventually run?

**Comment by josephaw1022** [comment]
It does eventually run but is a bit annoying. I normally only see this error while debugging and not while 'run without debugging'

**Comment by salarvand** [comment]
Yes, I tried it more an found out occurs in debugging mode and it is ok when you run without debugging.

**Comment by davidfowl** [comment]
This is a known issue that the debugging is breaking even for first chance exceptions on background threads.

**Comment by yudielcurbelo** [comment]
I'm getting the same error with just using `dotnet watch`. The error goes away with `Thread.Sleep(ms)`. It looks like the container is not fully active when trying to access it.
